cfme/utils/ftp.py

```diff
@@ -172,10 +178,10 @@ def path(self):
             whole path for this file
 
         """
-        if self.parent_dir:
+        if isinstance(self.parent_dir, FTPDirectory):
             return self.parent_dir.path + self.name
         else:
-            return self.name
+            return os.path.join(self.parent_dir, self.name)
```

Comment:
Is this doing the same thing as L182 above, joining the parent and current name?

Now is a good time to change the `+` concatenation on L182 as well, and use `'{}{}'.format(self.parent_dir.path, self.name)`